cupyx/scipy/sparse/_index.py

```diff
@@ -59,6 +58,51 @@ def _get_csr_submatrix(Ap, Aj, Ax,
     return Bp, Bj, Bx
 
 
+_csr_row_index_ker = core.ElementwiseKernel(
+    '''raw I rows, raw I Ap, raw I Aj, raw T Ax, raw I Bp''',
+    '''raw I Bj, raw T Bx''', '''
+    const I row = rows[i];
+    const I row_start = Ap[row];
+    const I row_end = Ap[row+1];
+
+    I out_row_idx = Bp[i];
+
+    // Copy columns
+    for(I j = row_start; j < row_end; j++) {
+        Bj[out_row_idx] = Aj[j];
+        Bx[out_row_idx] = Ax[j];
+        out_row_idx++;
+    }
+''', 'csr_row_index_ker', no_return=True)
```

Comment:
This kernel looks to work slow in case of the sparse matrix that has a few dense rows, like Leslie matrix..., but it's OK to leave the current implementation int the PR. We may work on this issue after the PR.

Reply:
Would the case of only a few rows and massive number of columns be a candidate for CSC instead of CSR or would that create other problems / bottlenecks in the workflow? 

I've been thinking of a way to remove the need for iterating over the rows by building a mask in parallel. It would increase the memory overhead, but it would definitely be nice to remove the need for iterating over each row.

Reply:
I think I've figured out a way to do this highly parallel. If I turn the `rows` array into a COO then I can use that, along with the output `Bp` to populate `Aj`, and `Ax` in parallel. I wouldn't mind trying this approach, so long as it won't hold up the release.